internal/osutils/autostart/autostart_linux.go

```diff
@@ -5,76 +5,31 @@ import (
 	"path/filepath"
 	"strings"
 
-	"github.com/ActiveState/cli/internal/assets"
 	"github.com/ActiveState/cli/internal/constants"
 	"github.com/ActiveState/cli/internal/errs"
 	"github.com/ActiveState/cli/internal/fileutils"
 	"github.com/ActiveState/cli/internal/osutils"
-	"github.com/ActiveState/cli/internal/osutils/shortcut"
 	"github.com/ActiveState/cli/internal/osutils/user"
 	"github.com/ActiveState/cli/internal/subshell/sscommon"
 )
 
 const (
-	autostartDir  = ".config/autostart"
 	autostartFile = ".profile"
 )
 
 func enable(exec string, opts Options) error {
+	if err := legacyDisableOnDesktop(exec, opts); err != nil {
+		return errs.Wrap(err, "Could not properly disable autostart (desktop): %v", err)
+	}
```

Comment:
This feels like the wrong place. Can invoke this logic through the installer instead? So it's removed at install time, and not during regular use.

Reply:
This is a Linux-specific behavior and the placement allows it to not affect the outer cross-platform surface. If that is, in fact, desirable, each OS will need to have an implementation for a new function in autostart.go.

Alternately, if the relevant file exists, we can report it via autostartPath, but that feels even more janky in that it relies on uninstallation for cleanup to occur. Though, I may be misunderstanding the expected flow.

In most cases, the current approach will look for a file, not find it, then take no action - so it's low cost and should, in more cases, ensure the file system is as we intend.

This seems complex enough that we might want to have a quick VC (along with @MDrakos) so that everyone is on the same page with the logic.

Reply:
There are legacy .profile mod cleanups and they fire in roughly the same places as the legacy desktop removals. The .profile mod cleanups have been moved into their own legacy function and are documented in the same Jira story as the legacy desktop removals.

Reply:
I'm not following why my requested change can't be addressed. My point is that this is effectively install/update logic, and should live in the installer and update code. This code already has platform specific logic, and even if not we can always introduce it. Go makes this dead simple so not sure I follow why this would require some larger refactor.